stuf_zds_payments/plugin.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from typing import Any
+
+from django.template.loader import render_to_string
+from django.utils.translation import gettext_lazy as _
+
+from openforms.registrations.contrib.stuf_zds.plugin import (
+    StufZDSRegistration,
+    ZaakOptionsSerializer,
+)
+from openforms.registrations.registry import register
+from openforms.submissions.models import Submission
+from openforms.template import render_from_string
+from openforms.template.validators import DjangoTemplateValidator
+from openforms.variables.service import get_static_variables
+from rest_framework import serializers
+from stuf.stuf_zds.client import ZaakOptions
+
+from .client import get_client
+from .registration_variables import register as variables_registry
+
+logger = logging.getLogger(__name__)
+
+PLUGIN_IDENTIFIER = "stuf-zds-create-zaak:ext-utrecht"
+
+
+def get_payment_status_update_text() -> str:
+    return render_to_string(
+        "registrations/contrib/stuf_zds_payments/payment_status_update_xml.txt"
+    ).strip()
+
+
+def prepare_value(value: Any):
+    match value:
+        case bool():
+            return "true" if value else "false"
+        case list():
+            return " ".join(value)
```

Comment:
have you looked at `stuf_zds.utils.flatten_data`? if using extra-elementen,  this should already handle dicts/lists, so a list like `["foo", "bar"]` would turn into:

```xml
<extraElement naam="list.0">foo</extraElement>
<extraElement naam="list.1">bar</extraElement>
```